File: src/main/java/com/baidu/algorithm/binarysearch/_153_Find_Minimum_in_Rotated_Sorted_Array.py
class Solution(object):
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0

        start, end = 0, len(nums) - 1
        while start < end:
            mid = (start + end) >> 1
            if nums[mid] > nums[end]:
                start = mid + 1
            else:
                end = mid

        return nums[start]

File: src/main/java/com/baidu/algorithm/binarysearch/_153_Find_Minimum_in_Rotated_Sorted_Array.java
/*
 * Copyright (C) 2018 Baidu, Inc. All Rights Reserved.
 */
package com.baidu.algorithm.binarysearch;

/**
 * _153_Find_Minimum_in_Rotated_Sorted_Array
 *
 * @author xuhaoran01
 */
public class _153_Find_Minimum_in_Rotated_Sorted_Array {

    public int findMin(int[] nums) {
        if(nums == null || nums.length == 0) {
            return 0;
        }

        int left = 0, right = nums.length - 1;
        while(left < right) {
            int mid = (right - left) / 2 + left;
            if(nums[mid] > nums[right]) {
                left = mid + 1;
            }
            else {
                right = mid;
            }
        }

        return nums[left];
    }
}
